src/content/1.7/modules/concepts/hooks/use-hooks-on-modern-pages.md

```diff
@@ -117,49 +86,138 @@ class ProductRepository
      */
     public function findAllbyLangId(int $langId)
     {
-        $prefix = $this->databasePrefix;
+        $prefix = $this->dbPrefix;
         $productTable = "${prefix}product";
         $productLangTable = "${prefix}product_lang";
 
         $query = "SELECT p.* FROM ${productTable} p LEFT JOIN ${productLangTable} pl ON (p.`id_product` = pl.`id_product`) WHERE pl.`id_lang` = :langId";
         $statement = $this->connection->prepare($query);
         $statement->bindValue('langId', $langId);
         $statement->execute();
-        
+
         return $statement->fetchAll();
     }
 }
 ```
 
-And declare your repository as a service:
+And declare your repository as a service. By convention, the declaration the name of the service is `your_society.module.your_module_name.your_service_name`, here it could be `mysociety.module.foo.product_repository`:
 
 ```yaml
-# modules/foo/config/services.yml
-
 services:
-    product_repository:
-        class: Foo\Repository\ProductRepository
-        arguments: ['@doctrine.dbal.default_connection', '%database_prefix%']
+  _defaults:
+    public: true
+
+  <your_society>.module.<your_module_name>.<your_service_name>:
+    class: PrestaShop\Module\Foo\Repository\ProductRepository
+    arguments:
+      $connection: '@doctrine.dbal.default_connection'
+      $dbPrefix: '%database_prefix%'
 ```
 
 You can now use it in your module (and everywhere in PrestaShop modern pages!):
 
+### Third step: create and register the Hook
+
+Create a [new module](http://doc.prestashop.com/display/PS17/Creating+a+first+module) called `foo` and register the hook. You should end up with this kind of code in your module:
+
 ```php
-// foo.php
+<?php
 
-/* ... */
-/**
- * Get the list of products for a specific lang.
- */
-public function hookDisplayDashboardToolbarIcons($hookParams)
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+if (file_exists(__DIR__ . '/vendor/autoload.php')) {
+    require_once __DIR__ . '/vendor/autoload.php';
+}
+
+use PrestaShop\Module\Foo\Repository\ProductRepository;
+use PrestaShop\PrestaShop\Adapter\SymfonyContainer;
+
+class Foo extends Module
 {
-    if ($this->isSymfonyContext() && $hookParams['route'] === 'admin_product_catalog') {
-        $products = $this->get('product_repository')->findAllByLangId(1);
-        dump($products);
+    public function __construct()
```

Comment:
This code is now a bit long, you could reduce it using `...` inside the class
The idea is not to provide a fully functional class to copy, but to emphasize the code related to what you are explaining

So in this part the constructor, and the install/uninstall methods are less relevant. You should focus on the hook code, by the way nice display of the fallback service creation